text/0000-introduce empty ranges.md

```diff
@@ -0,0 +1,255 @@
+Range(0, inf, inf): .is_empty() = false, .is_infinite() = true
+Range(0, -inf, 0): .is_empty() = false, .is_infinite() = true
+Range(0, -inf, -10): .is_empty() = false, .is_infinite() = true
+Range(0, -inf, -inf): .is_empty() = false, .is_infinite() = true
+Range(inf, 0, 0): .is_empty() = false, .is_infinite() = true
+Range(inf, 0, -10): .is_empty() = false, .is_infinite() = true
+Range(inf, 0, -inf): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, 0): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, 10): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, inf): .is_empty() = false, .is_infinite() = true
+Range(inf, 0, nan): .is_empty() = false, .is_infinite() = true
+Range(0, inf, nan): .is_empty() = false, .is_infinite() = true
+Range(-inf, 0, nan): .is_empty() = false, .is_infinite() = true
+Range(0, -inf, nan): .is_empty() = false, .is_infinite() = true
+Range(nan, 0, inf): .is_empty() = false, .is_infinite() = true
+Range(0, nan, inf): .is_empty() = false, .is_infinite() = true
+Range(nan, 0, -inf): .is_empty() = false, .is_infinite() = true
+Range(0, nan, -inf): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, 1): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, -1): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, 0): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, nan): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, inf): .is_empty() = false, .is_infinite() = true
+Range(inf, inf, -inf): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, 1): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, -1): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, 0): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, nan): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, inf): .is_empty() = false, .is_infinite() = true
+Range(-inf, -inf, -inf): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, 0): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, 1): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, inf): .is_empty() = false, .is_infinite() = true
+Range(nan, nan, -inf): .is_empty() = false, .is_infinite() = true
+```
+
+# How We Teach This
+
+This can be taught by improving the documentation of `Range` and by discussing a few examples in the tutorial section. Also, as stated above, `Range.is_infinite()` is hardly used on Github and the impact of this change should be considered mild.
+
+Tests can easily be added to `packages/collections/_test.pony` that would add a `_assert_empty` method and test for relevant cases while removing thoses cases from the current `_assert_infinite` tests.
+
+# Drawbacks
+
+* As far as I can tell from public code, Range.is_infinite() is not used in any Pony code other than test code associated with tests for Range.
+* This should not produce any bugs in the compiler since it is a small library change only
+* Maintenance cost should be marginal.
```

Comment:
If you are going to make a statement about maintenance cost, it should have at least some justification. Given this is in drawbacks, it sounds like you believe there is an additional maintenance cost vs the status-quo. If that is the case, it should be detailed here. If there isn't an additional maintenance cost, that should be discussed elsewhere, not in drawback.